autogpt_platform/frontend/src/components/cron-scheduler.tsx

```diff
@@ -0,0 +1,460 @@
+import React, { useEffect, useState } from "react";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { useToast } from "@/components/ui/use-toast";
+import { Separator } from "@/components/ui/separator";
+import { makeCronExpression } from "@/lib/cron-expression-utils";
+import { Dialog, DialogContent, DialogTitle } from "@/components/ui/dialog";
+
+type CronFrequency =
+  | "minute"
+  | "hour"
+  | "daily"
+  | "weekly"
+  | "monthly"
+  | "yearly"
+  | "custom";
+
+export function CronScheduler({
+  onCronExpressionChange,
+}: {
+  onCronExpressionChange: (cronExpression: string) => void;
+}): React.ReactElement {
+  const [frequency, setFrequency] = useState<CronFrequency>("daily");
+  const [selectedDays, setSelectedDays] = useState<number[]>([]);
+  const [selectedTime, setSelectedTime] = useState<string>("09:00");
+  const [showCustomDays, setShowCustomDays] = useState<boolean>(false);
+  const [selectedMinute, setSelectedMinute] = useState<string>("0");
+  const [customInterval, setCustomInterval] = useState<{
+    value: number;
+    unit: "minutes" | "hours" | "days";
+  }>({ value: 1, unit: "minutes" });
+
+  // const [endType, setEndType] = useState<"never" | "on" | "after">("never");
+  // const [endDate, setEndDate] = useState<Date | undefined>();
+  // const [occurrences, setOccurrences] = useState<number>(1);
+
+  const weekDays = [
+    { label: "Su", value: 0 },
+    { label: "Mo", value: 1 },
+    { label: "Tu", value: 2 },
+    { label: "We", value: 3 },
+    { label: "Th", value: 4 },
+    { label: "Fr", value: 5 },
+    { label: "Sa", value: 6 },
+  ];
+
+  const months = [
+    { label: "Jan", value: "January" },
+    { label: "Feb", value: "February" },
+    { label: "Mar", value: "March" },
+    { label: "Apr", value: "April" },
+    { label: "May", value: "May" },
+    { label: "Jun", value: "June" },
+    { label: "Jul", value: "July" },
+    { label: "Aug", value: "August" },
+    { label: "Sep", value: "September" },
+    { label: "Oct", value: "October" },
+    { label: "Nov", value: "November" },
+    { label: "Dec", value: "December" },
+  ];
+
+  useEffect(() => {
+    const cronExpr = makeCronExpression(
+      frequency,
+      selectedTime,
+      selectedDays,
+      selectedMinute,
+      customInterval,
+    );
+    onCronExpressionChange(cronExpr);
+  }, [
+    frequency,
+    selectedTime,
+    selectedDays,
+    selectedMinute,
+    customInterval,
+    onCronExpressionChange,
+  ]);
+
+  return (
+    <div className="max-w-md space-y-6">
+      <div className="space-y-4">
+        <Label className="text-base font-medium">Repeat</Label>
+
+        <Select
+          onValueChange={(value: CronFrequency) => setFrequency(value)}
+          defaultValue="daily"
+        >
```

Comment:
This should be a fully controlled input:
```tsx
<Select value={frequency} onValueChange={setFrequency}>
```
Otherwise, we are mixing controlled/uncontrolled patterns on this input, which could lead to problems down the line 💭